### DISCUSSION TITLE
Communication Timeouts increased after replacing controller board

### DISCUSSION THREAD

**Discussion by ebzimny01** [jgyates/genmon#1064] [Questions] [answered]
My generator had an under voltage error and was only outputting 75V. The repair required replacement of the control board. It's a 22kw Generac with Evolution 1 controller. It was replaced with same controller. After this new controller, GenMon is repeatedly cycling through alerts "Not receiving data" and then "Health OK" (via Slack integration). Looking at monitor data, it shows 17-20% timeout errors under Communication Stats. Any advice for troubleshooting?

**Comment by jgyates** [comment]
are you using serial or serial over TCP? If serial over TCP are you using WiFi?

**Reply by ebzimny01** [reply]
The pi is directly connected to the controller, I believe via serial, and housed inside my generator. The pi is connected to my network via WiFi.

**Reply by jgyates** [reply]
You do not include any info about your generator model or what version of genmon you are running. So first, I would make sure you are running the latest version of the software by updated your software from the About page. Once the software updates and restarts then I would follow the setps in the link below for serial troubleshooting. If you have not made any changes to the pi since it was working with your last controller then you can skip to the section named "Run a  Loopback" on the page below.

     https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting

**Reply by ebzimny01** [reply]
Noted - it is Generac 22kw Model : G0070431. I was on GenMon v1.19.01 yesterday, but updated it to v1.19.02 after the controller was replaced this morning and seeing the comm problem. I did not do the controller repair. Let me open it back up, check the connections and will checkout the troubleshooting link . . .

**Reply by skipfire** [reply]
@ebzimny01 did you build your own cable or buy a HAT?  If you built your own, I would pay the most attention to ground, especially since you said your controller got replaced.  There are 2 grounds in the Molex, and sometimes one of them is flaky or disconnected so you may need to move 2 pins over to use the other ground.

**Reply by ebzimny01** [reply, marked as answer]
Definitely an issue with serial comms after running the serial tests. Looks like my cable is a combination of a DB9 connector plus [2 of these](https://www.amazon.com/gp/product/B000M802RG) connected together. The DB9 and RS232 serial connection to the Pi looks good. I may just replace those 2 molex cables and see if that resolves the issue.

**Comment by jgyates** [comment]
Thanks, this status update helps others who may have the same type of issues. Let us know how it goes.

**Reply by ebzimny01** [reply]
Ultimately I decided to purchase the USB to RS232 to Molex adapter [here](https://pintsize.me/store/ols/products/molex-to-usb-converter) and switch GenMon to use \dev\ttyUSB0 as the serial port. My GenMon is now functioning again.

**Comment by jgyates** [comment]
I am glad it is working for you. Thanks for the update.
